xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumer.java

```diff
@@ -181,42 +199,48 @@ private void traverseXObjectsOnCreate(Map<DocumentReference, List<BaseObject>> x
     /**
      * Handles an object of a created document and search for mentions to notify its content.
      * @param baseObject the object
-     * @param documentReference the reference of the document
+     * @param version the version of the document holding the base object when it was created
      * @param authorReference the reference of the author of the document
      * @param syntax the syntax of the document
+     * @param mentionNotificationParameters hold the list of identified mentions
```

Comment:
We agree that this map contains the mentions we already know about to compare if the ones we traverse are new?

Reply:
This store a list of `MentionNotificationParameters`. They are created for each traversed entity, regardless of if they have new mentions.
A the end of the analysis, we only send events for the `MentionNotificationParameters` holding new mentions.